### PR TITLE
New google operator: SQLToGoogleSheetsOperator

### PR DESCRIPTION
**Pull request by mariotaddeucci**
Adds an operator to send sql results to google spreadsheet.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/google/suite/transfers/sql_to_sheets.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    def _data_prep(self, data):
+        for row in data:
+            item_list = []
+            for item in row:
+                if type(item) is datetime.date:
+                    item = item.strftime("%Y-%m-%d")
+                elif type(item) is datetime.datetime:
+                    item = item.strftime("%Y-%m-%d")
+                elif type(item) is decimal.Decimal:
+                    item = float(item)
```
Comment (uranusjr) [inline airflow/providers/google/suite/transfers/sql_to_sheets.py]: This `type(item) is <type>` kind of check is really not a good idea. Also you dropped the time part in a datetime, is it intended?

Something like this is likely better:

```suggestion
                if isinstance(item, (datetime.date, datetime.datetime)):
                    item = item.isoformat()
                elif isinstance(item, int):  # To exclude int from the number check.
                    pass
                elif isinstance(item, number.Number):
                    item = float(item)
```

File: airflow/providers/google/suite/transfers/sql_to_sheets.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        with closing(hook.get_conn()) as conn:
+            with closing(conn.cursor()) as cur:
+                self.log.info("Executing query")
+                if self.parameters is not None:
+                    cur.execute(self.sql, self.parameters)
+                else:
+                    cur.execute(self.sql)
+
+                yield [field[0] for field in cur.description]
+                yield from self._data_prep(cur.fetchall())
```
Comment (uranusjr) [inline airflow/providers/google/suite/transfers/sql_to_sheets.py]:
```suggestion
        with closing(hook.get_conn()) as conn, closing(conn.cursor()) as cur:
            self.log.info("Executing query")
            cur.execute(self.sql, self.parameters or ())

            yield [field[0] for field in cur.description]
            yield from self._data_prep(cur.fetchall())
```

File: airflow/providers/google/suite/transfers/sql_to_sheets.py (hunk 1 above)
Comment (mariotaddeucci) [reply airflow/providers/google/suite/transfers/sql_to_sheets.py]: The dropped time was a mistake, fixed it now with your suggestion :)

File: airflow/providers/google/suite/transfers/sql_to_sheets.py (hunk 2 above)
Comment (mariotaddeucci) [reply airflow/providers/google/suite/transfers/sql_to_sheets.py]: Awesome

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by mariotaddeucci** [comment]
Woohoo!
Thanks @uranusjr and @potiuk o/

**Comment by ashb** [comment]
Hmmm, curious, this seems to be causing the docs build to fail on mster, but it passed on this PR branch.

**Comment by potiuk** [comment]
I think the failing builds are something else entirely - looks like `eager-upgrade` in main is causing the errors by some new dependencies (will take a look tomorrow)

**Pull request merged**